python/GafferUITest/TestCase.py

```diff
@@ -83,6 +83,21 @@ def assertExampleFilesExist( self ) :
 			expanded = os.path.expandvars( e['filePath'] )
 			self.assertTrue( os.path.exists( expanded ), "%s does not exist" % expanded )
 
+	def assertExampleFileDontReferenceUnstablePaths( self ) :
+
+		forbidden = (
+			"${scriptName}",
+			"/home"
+		)
+
+		examples = GafferUI.Examples.registeredExamples()
+		for e in examples.values():
+			path = os.path.expandvars( e['filePath'] )
+			with open( path, 'r' ) as example :
+				for line in example :
```

Comment:
We won't be able to ever use the substrings `${scriptName}` or `/home` (and any other verboten substrings we think of later) in backdrops' `description` plugs, annotations, or other plug/nodes metadata. Can we check only for `filename` plugs?

Reply:
Addressed in https://github.com/GafferHQ/gaffer/pull/3309/commits/ae141bf1daad86961ce23c67970679d35e78d5ed. There are numerous plugs where this can cause problems, so I went for white-listing safe plugs instead. We can always work on adding others or changing the logic as needed if we encounter problematic situations.